test/unit/local/steps/await_write_finish.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-env mocha */
+/* @flow */
+
+const should = require('should')
+const awaitWriteFinish = require('../../../../core/local/steps/await_write_finish')
+const Buffer = require('../../../../core/local/steps/buffer')
+
+describe('core/local/steps/await_write_finish', () => {
+  it('should reduce created→deleted to empty', async () => {
+    const buffer = new Buffer()
+    const originalBatch = [
+      {
+        action: 'created',
+        kind: 'file',
+        path: __filename
+      },
+      {
+        action: 'deleted',
+        kind: 'file',
+        path: __filename
+      }
+    ]
+    originalBatch.forEach(event => {
+      buffer.push([Object.assign({}, event)])
+    })
+    const enhancedBuffer = awaitWriteFinish(buffer, {})
+    should(enhancedBuffer._buffer).be.empty()
```

Comment:
Could we avoid checking the internals of `Buffer`? Adding another event to the buffer would let us test that only this specific event is returned by `pop()`.

Reply:
To check if the `buffer` is empty, I `push` an event, and `pop` it and check that it is the same?
Ok, that seems not very explicit, but I can do that.

Reply:
That's the idea, yes. I know it feels forced but at least it uses only the public API of the module.

Reply:
Another option could be to add an `empty` method on the `Buffer` class

Reply:
s/empty/isEmpty/ ;-)

Reply:
Will be [refactored in an upcoming PR](https://github.com/cozy-labs/cozy-desktop/pull/1402#issuecomment-455204471)